src/new/wallet.js

```diff
@@ -1903,7 +1907,18 @@ class HathorWallet extends EventEmitter {
     }
 
     // Always create another melt authority output
-    const ret = tokens.generateMeltData(meltInput[0], tokenUid, amount, pin, newOptions.createAnotherMelt, { depositAddress: newOptions.address, changeAddress: newOptions.changeAddress });
+    const ret = tokens.generateMeltData(
+      meltInput[0],
+      tokenUid,
+      amount,
+      pin,
+      newOptions.createAnotherMelt,
+      {
+        depositAddress: newOptions.address,
+        changeAddress: newOptions.changeAddress,
```

Comment:
Maybe add an option `allowExternalAddress: true/false`? I guess it would prevent operational mistakes.

Reply:
Added the parameter with a different name because there are 3 addresses as parameters.

Reply:
I just checked the changes. I understand that there might be more than one address in several cases. In this case, I'd rather use an `allowExternalAddresses: string[]` (or a set) that contains a list of authorized external addresses to be used anywhere in the method.

The current code base is good and we can also move forward with it and change it later.

What do you think? Change now or later?

Reply:
We just had a conversation about it and I agree that the current implementation is better than my suggestion.